st2common/st2common/models/api/keyvalue.py

```diff
@@ -57,8 +76,39 @@ class KeyValuePairAPI(BaseAPI):
         'additionalProperties': False
     }
 
+    @staticmethod
+    def _setup_crypto():
+        LOG.info('Checking if encryption is enabled for key-value store.')
+        KeyValuePairAPI.is_encryption_enabled = cfg.CONF.keyvalue.enable_encryption
+        LOG.debug('Encryption enabled? : %s', KeyValuePairAPI.is_encryption_enabled)
+        if KeyValuePairAPI.is_encryption_enabled:
+            KeyValuePairAPI.crypto_key_path = cfg.CONF.keyvalue.encryption_key_path
+            LOG.info('Encryption enabled. Looking for key in path %s',
+                     KeyValuePairAPI.crypto_key_path)
+            if not os.path.exists(KeyValuePairAPI.crypto_key_path):
+                msg = ('Encryption key file does not exist in path %s.' %
+                       KeyValuePairAPI.crypto_key_path)
+                LOG.exception(msg)
+                LOG.info('All API requests will now send out BAD_REQUEST ' +
+                         'if you ask to store secrets in key value store.')
+                KeyValuePairAPI.crypto_key = None
+            else:
+                KeyValuePairAPI.crypto_key = KeyValuePairAPI._read_crypto_key(
```

Comment:
Unless I'm missing something it seems like this is called only once.

If that's the case, since the whole API has access to this part I think it would be safer to only load the key when needed. I know it will be slower than caching the key, but it means the key won't stuck in the memory when we don't need it (well, that's not 100% true since we don't do manual garbage collection, etc., but still).